paasta_tools/tron_tools.py

```diff
@@ -1093,23 +1093,41 @@ def load_tron_instance_config(
     load_deployments: bool = True,
     soa_dir: str = DEFAULT_SOA_DIR,
 ) -> TronActionConfig:
-    jobs = load_tron_service_config(
+    for action in load_tron_instance_configs(
         service=service,
         cluster=cluster,
         load_deployments=load_deployments,
         soa_dir=soa_dir,
-    )
-    requested_job, requested_action = instance.split(".")
-    for job in jobs:
-        if job.get_name() == requested_job:
-            for action in job.get_actions():
-                if action.get_action_name() == requested_action:
-                    return action
+    ):
+        if action.get_instance() == instance:
+            return action
     raise NoConfigurationForServiceError(
         f"No tron configuration found for {service} {instance}"
     )
 
 
+@time_cache(ttl=5)
```

Comment:
this could be a standard lru_cache, right? there's no long-running paasta code that would call this otherwise, is there?

(or does the paasta api call this for paasta status at some point?)

Reply:
Hmm, good question. lru_cache could definitely end up causing problems for any long-running code. I'm just doing the same thing we do for `load_tron_service_config`